pkg/providers/k8s/k8s.go

```diff
@@ -21,9 +23,66 @@ func NewProvider(pullSecretsGetter func(image string) []corev1.Secret) *Provider
 	}
 }
 
+func (p Provider) correctDockerRegistry(secrets []corev1.Secret) ([]corev1.Secret, error) {
+	for i, secret := range secrets {
+		if secret.Type != corev1.SecretTypeDockerConfigJson {
+			continue
+		}
+
+		data, exists := secret.Data[corev1.DockerConfigJsonKey]
+		if !exists {
+			continue
+		}
+
+		dockerConfig, err := parseDockerConfig(data)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse docker config for secret %d: %w", i, err)
+		}
+
+		if err := updateDockerRegistryAuths(dockerConfig); err != nil {
+			return nil, fmt.Errorf("failed to update docker registry auths for secret %d: %w", i, err)
+		}
+
+		updatedData, err := json.Marshal(dockerConfig)
+		if err != nil {
+			return nil, fmt.Errorf("failed to marshal updated docker config for secret %d: %w", i, err)
+		}
+
+		secrets[i].Data[corev1.DockerConfigJsonKey] = updatedData
+	}
+	return secrets, nil
+}
+
+func parseDockerConfig(data []byte) (map[string]interface{}, error) {
+	var dockerConfig map[string]interface{}
+	if err := json.Unmarshal(data, &dockerConfig); err != nil {
+		return nil, fmt.Errorf("unmarshalling docker config: %w", err)
+	}
+	return dockerConfig, nil
+}
+
+func updateDockerRegistryAuths(dockerConfig map[string]interface{}) error {
+	auths, ok := dockerConfig["auths"].(map[string]interface{})
```

Comment:
In "kubernetes.io/dockercfg" there is no auth, we have to parse these two types differently.